backvendor/gosource.go

```diff
@@ -187,52 +187,29 @@ func findImportComment(src *GoSource) (string, error) {
 		if err != nil {
 			return err
 		}
-		if info.IsDir() {
-			if info.Name() != "." &&
-				strings.HasPrefix(info.Name(), ".") {
-				return filepath.SkipDir
-			}
-			if info.Name() == "vendor" || info.Name() == "testdata" {
-				return filepath.SkipDir
-			}
+		if !info.IsDir() {
 			return nil
 		}
-		if !strings.HasSuffix(info.Name(), ".go") {
-			return nil
+		if info.Name() != "." &&
+			strings.HasPrefix(info.Name(), ".") {
+			return filepath.SkipDir
 		}
-		r, err := os.Open(pth)
-		if err != nil {
-			return err
+		switch info.Name() {
+		case "vendor", "testdata":
+			return filepath.SkipDir
 		}
-		defer r.Close()
-		scanner := bufio.NewScanner(bufio.NewReader(r))
-		for scanner.Scan() {
-			line := scanner.Text()
-			fields := strings.Fields(line)
-			if len(fields) < 5 {
-				continue
-			}
-			if fields[0] == "//" || fields[0] == "/*" {
-				continue
-			}
-			if fields[0] != "package" {
-				return nil
-			}
-			if fields[2] != "/*" && fields[2] != "//" {
-				return nil
-			}
-			if fields[3] != "import" {
-				return nil
-			}
-			pth := fields[4]
-			if len(pth) < 3 {
-				return nil
-			}
-			if pth[0] != '"' ||
-				pth[len(pth)-1] != '"' {
+
+		pkg, err := build.ImportDir(pth, build.ImportComment)
+		if err != nil {
+			if _, ok := err.(*build.NoGoError); ok {
 				return nil
 			}
-			importPath = pth[1 : len(pth)-1]
+			return err
+		}
+		if pkg.ImportComment != "" {
+			importPath = pkg.ImportComment
+			log.Debugf("found import path from import comment: %s",
+				importPath)
 			return errFound
```

Comment:
When you declare `errFound`, could you add a comment explaining that this is a way to end the `Walk` method early?

Reply:
Added a comment explaining this.